### PR TITLE
"Friends to the End" starting scenario with a guaranteed recruitable follower

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Content "Friends to the End scenario for DDA and DDotD, adding a unique guaranteed follower and a stub of a storyline."

#### Purpose of change
With all the NPC changes I've added, I think we need some ways to encourage playtesting of NPCs. There should be some way to start with an NPC companion.

We've also talked for years about having an NPC companion that offers some tutorial style guidance, and this could be a way to start the ball rolling on that.

#### Describe the solution
Adds Liam, a unique NPC companion that can begin the game as your follower. Liam will have some unique character traits, dialogue, and story. Ideally in the future, he might be something of a tutorial character that can give you tips for new players. For now, he'll basically be any other NPC but you can get him to be your friend without having to pass any speech checks and start right next to him.

Liam is going to be primarily a Dark Days of the Dead character, and I'd like to make this the default start for DDotD. I will still keep him fleshed out for mainline, but I may leave the mainline tutorials for someone else. For DDotD he'll explain the game's mechanics. If someone wants to add similar tutorial content to the main game, he should be good for that.

- This is a very "beginner friendly" scenario. It starts you with some decent weapons, a strong NPC follower, and a functional vehicle. I've made it this way because I want it to become a good alternative start: eventually, Liam should have more quests to get you started, analogous to the refugee center quest.
- I believe this may be the first scenario to start with a bit of lore. I imagine this could be expanded in the future. Currently you get an extra dialogue describing the situation when you open the game. Later, I'd love to have some options in there where you can explain who Liam is to you, and have his character change based on your choices.
- I don't think I'll add it in this PR, but I'd very much like to have a couple more starting missions in here, where you have to take Chris's body out and bury it and Liam gives you tips through the process.

#### Describe alternatives you've considered
Infinite possible, but for NPC playtesting needs, I really have to get people to get a follower right off the bat.
I have considered giving Liam some "META" tagged conversation that explains what playtests I need, but I don't think there's a lot of need for it yet.

#### Testing
Tested in vanilla and ddotd.

#### Additional context
In the future I would like to have more of these types of NPCs, where you'd select the "start with a friend" scenario (name pending) and then choose location, and each location would have a different friend.

### CODE REVIEW

File: data/json/overmap/overmap_special/specials.json [hunk 1]
```diff
@@ -634,6 +634,16 @@
...
   },
+  {
+    "type": "overmap_special",
+    "id": "cabin_liam",
+    "overmaps": [ { "point": [ 0, 0, 0 ], "overmap": "cabin_liam_north" }, { "point": [ 0, 0, 1 ], "overmap": "cabin_roof_1_north" } ],
+    "locations": [ "forest" ],
+    "city_distance": [ 25, -1 ],
+    "city_sizes": [ 0, 20 ],
+    "occurrences": [ 0, 0 ],
+    "flags": [ "CLASSIC", "WILDERNESS", "UNIQUE", "MAN_MADE" ]
```
Comment (I-am-Erk) [inline data/json/overmap/overmap_special/specials.json:645]:
```suggestion
    "flags": [ "CLASSIC", "WILDERNESS", "UNIQUE", "MAN_MADE", "SHOULD_NOT_SPAWN" ]
```

**Comment by Maddremor** [comment]
Perhaps not strictly in the scope of this PR, but a background that makes the NPC you spawn alongside have a 100% recruitment chance could be a thing and would slot into any existing scenario where you spawn along someone else.

File: data/json/overmap/overmap_terrain/overmap_terrain.json [hunk 2]
```diff
@@ -242,6 +242,17 @@
...
+  {
+    "type": "overmap_terrain",
+    "id": [ "cabin_liam" ],
+    "name": "cabin",
+    "sym": "C",
+    "color": "i_green",
+    "see_cost": 5,
+    "extras": "build",
+    "mondensity": 2,
+    "flags": [ "RISK_LOW", "SOURCE_SAFETY", "SOURCE_FOOD", "SHOULD_NOT_SPAWN" ]
```
Comment (I-am-Erk) [inline data/json/overmap/overmap_terrain/overmap_terrain.json:254]:
```suggestion
    "flags": [ "RISK_LOW", "SOURCE_SAFETY", "SOURCE_FOOD" ]
```

File: data/mods/classic_zombies/npcs/liam_follower.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "responses": [ { "text": "So Chris turned into a zombie right in front of you?", "topic": "TALK_DDotD_Liam_Meeting3" } ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_DDotD_Liam_Tut1_bit",
+    "dynamic_line": "You got it.  I know, it feels really contrived, doesn't it?  It's exactly like the old movies in every way, like someone had designed it to be.  I don't know if it's every bite or what, but Chris just had one that barely broke the skin and you see where that ended up.  Took about three days.  My advice is, we don't let these things teeth anywhere near our skin.  They're pretty clumsy bastards, though.  They definitely can't just nibble you as they go by, they've gotta grab you first.  Unfortunately, seems like they're pretty fuckin' good at that.  Once you get grabbed, best bet is to go full ham on the zombie grabbing you with everything you've got, and try to shake loose, but honestly it's better to not get grabbed.",
+    "responses": [ { "text": "Well, no bites. I'm fine.  Want to get this cleaned up?", "topic": "TALK_DDotD_Liam_Meeting4" } ]
```
Comment (I-am-Erk) [inline data/mods/classic_zombies/npcs/liam_follower.json:51]:
```suggestion
    "responses": [ { "text": "Well, no bites.  I'm fine.  Want to get this cleaned up?", "topic": "TALK_DDotD_Liam_Meeting4" } ]
```

**Pull request merged**